MHCollapsibleViewManager.m

```diff
@@ -0,0 +1,228 @@
+//
+//  MHCollapsibleViewManager.m
+//  This is a manager of a uiview and view controller
+//
+//  Created by Lizzy Randall on 5/6/15.
+//  Copyright (c) 2015 Lizzy Randall. All rights reserved.
+//
+
+#import "MHCollapsibleViewManager.h"
+#import "MHFilterLabel.h"
+
+@implementation MHCollapsibleViewManager
+
+//@synthesize tableView;
+//@synthesize viewController;
+@synthesize rowAnimation;
+
+@synthesize data;
+@synthesize cellStyles;
+@synthesize headerTitle;
+
+@synthesize expanded;
+@synthesize numOfRows;
+
+//Initialize with animation, title and tableView
+- (id)initalizeManager:(UITableViewRowAnimation)animation title:(NSString*) title tableView:(UITableView*)tableView
+{
+    self.headerTitle = title;
+    self.rowAnimation = animation;
+    //initialized first and then can be overwritten by controller
+    //tableView is needed to create cells to store types
+    [self setCellProperties:nil header:nil clickedCell:nil normalCell:nil tableView:tableView];
+    return self;
+}
+
+//called after Initializing Manager
+- (void) initializeData:(NSMutableArray*)dataFromController
+{
+    self.data = [[NSMutableArray alloc] init];
+    for(int i=0; i < [dataFromController count] ; i++)
+    {
+        [data addObject:[MHFilterLabel constructLabel:[dataFromController objectAtIndex:i] checked:false]];
+    }
+    self.numOfRows = (NSInteger)[data count];
+}
+
+//set the cell provided with same settings as corresponding array type
+- (UITableViewCell*)setCellSettings:(UITableViewCell*)cell type:(NSString*)type
+{
+    UITableViewCell *tempCell = nil;
+    int index = 3; //normall cell
+    if([type isEqual: @"ClickedHeader"])
+    {
+        index = 0;
+    }
+    else if([type isEqual: @"NormalHeader"])
+    {
+        index = 1;
+    }
+    else if([type isEqual: @"ClickedCell"])
+    {
+        index = 2;
+    }
+    
+    tempCell = [cellStyles objectAtIndex:index];
+    cell.accessoryType = tempCell.accessoryType;
+    cell.accessoryView = tempCell.accessoryView;
+    cell.textLabel.textColor = tempCell.textLabel.textColor;
+    
+    if(index == 0 || index == 1)
+    {
+        cell.textLabel.text = self.headerTitle;
+        if(self.numOfRows > 0)
+            cell.detailTextLabel.text = [NSString stringWithFormat:NSLocalizedString(@"%i items",), self.numOfRows];
+        else
+            cell.detailTextLabel.text = @"No item";
+    }
+    tempCell = nil;
+    return cell;
+}
+
+//takes in four cells that are formatted or takes nil and creates default cells
+- (void)setCellProperties:(UITableViewCell*)clickedHeader header:(UITableViewCell*)header clickedCell:(UITableViewCell*)clickedCell normalCell:(UITableViewCell*)cell tableView:(UITableView*)tableView
```

Comment:
A further abstraction I would recommend as your example gets more complex is to move the cell styling from a method like this into a subclass of UITableCell. You could even just move this method into that subclass.
If you have multiple types of cells that you need for different type of data you could either have multiple configuration methods like this one or multiple subclasses (if you have multiple subclasses you need to make sure you get the right kind of object from the reuse queue that the tableView provides.